pgmpy/tests/test_prediction/test_NaiveAdjustmentRegressor.py

```diff
@@ -0,0 +1,407 @@
+#     #         0         1         2
+#     # 0  0.466932  0.807233  0.867482
+#     # 1  0.824674  0.404883  0.821456
+#     # 2  0.313492  0.961640  0.841893
+#     # 3  0.805902  0.938890  0.425192
+#     # 4  0.587333  0.477575  0.309112
+#     # Column names: [0, 1, 2]
+#     # Column types: [<class 'int'>, <class 'int'>, <class 'int'>]
+#     dag = DAG(
+#         ebunch=[],  # No edges needed for simplest case
+#         roles={
+#             "exposure": [0],
+#             "outcome": ["target"],
+#             "adjustment": []
+#         },
+#     )
+#     return NaiveAdjustmentRegressor(causal_graph=dag, estimator=LinearRegression())
+
+
+def make_estimator():
+    """Create a valid estimator for sklearn compatibility tests."""
+    dag = DAG()
+    dag.add_nodes_from([0, 1])  # exposure=0, outcome=1 (dummy)
+
+    # assign roles
+    dag = dag.with_role("exposure", [0])
+    dag = dag.with_role("outcome", [1])
+    dag = dag.with_role("adjustment", [])
+    return NaiveAdjustmentRegressor(causal_graph=dag, estimator=LinearRegression())
+
+
+@parametrize_with_checks([make_estimator()])
+def test_sklearn_compatibility(estimator, check):
+    """Test sklearn compatibility using parametrize_with_checks."""
+    check(estimator)
+
+
+def test_basic_functionality_with_adjustment():
+    """Test basic fit and predict functionality with synthetic causal data."""
+
+    lgbn = DAG.from_dagitty(
+        "dag { Z -> X [beta=0.5] X -> Y [beta=2.0] Z -> Y [beta=1.5] }"
+    )
+
+    data = lgbn.simulate(1000, seed=42)
+
+    dag = DAG(
+        ebunch=[("Z", "X"), ("Z", "Y"), ("X", "Y")],
+        roles={"exposure": "X", "outcome": "Y", "adjustment": ["Z"]},
+    )
+
+    estimators = [
+        LinearRegression(),
+        RandomForestRegressor(n_estimators=10, random_state=42),
+        DummyRegressor(),
+    ]
+
+    for base_est in estimators:
+        regressor = NaiveAdjustmentRegressor(causal_graph=dag, estimator=base_est)
+
+        train_size = int(0.7 * len(data))
+        X_train = data[["X", "Z"]].iloc[:train_size]
+        y_train = data["Y"].iloc[:train_size]
+        X_test = data[["X", "Z"]].iloc[train_size:]
+        y_test = data["Y"].iloc[train_size:]
+
+        regressor.fit(X_train, y_train)
+        predictions = regressor.predict(X_test)
+
+        assert len(predictions) == len(y_test)
+        assert regressor.exposure_var_ == "X"
+        assert regressor.outcome_var_ == "Y"
+        assert regressor.adjustment_vars_ == ["Z"]
+
+        if isinstance(base_est, LinearRegression):
+            # Expected coefficients: [X_coef=2.0, Z_coef=1.5] from simulation
+            # Feature order is [X, Z] (exposure + adjustment)
+            expected_coefs = [2.0, 1.5]  # beta coefficients from DAG simulation
+            actual_coefs = regressor.estimator_.coef_
+
+            # Check coefficients are close to expected values (allowing for noise)
+            np.testing.assert_allclose(actual_coefs, expected_coefs, atol=0.2)
```

Comment:
`0.2` is an extremely high tolerance for this simple graph. Can you try `0.05` here?